### PR TITLE
Telemetry guide

### PR DESCRIPTION
**Pull request by mcrumm**
A guide to telemetry events and the Telemetry supervisor.

### CODE REVIEW

File: guides/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+currently emitting `:telemetry` events.
+
+Library authors are actively encouraged to send a PR adding
+their own (in alphabetical order, please):
+
+* [Absinthe](https://hexdocs.pm/absinthe) - Coming Soon!
+* [Broadway](https://hexdocs.pm/broadway) - [Events](https://hexdocs.pm/broadway/Broadway.html#module-telemetry)
+* [Ecto](https://hexdocs.pm/ecto) - [Events](https://hexdocs.pm/ecto/Ecto.Repo.html#module-telemetry-events)
+* [Oban](https://hexdocs.pm/oban) - [Events](https://hexdocs.pm/oban/Oban.Telemetry.html)
+* [Phoenix](https://hexdocs.pm/phoenix) - [Events](https://hexdocs.pm/phoenix/Phoenix.Endpoint.html#module-instrumentation)
```
Comment (josevalim) [inline guides/telemetry.md:387]: I believe we move them to `Phoenix.Logger`.

File: guides/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+on `:telemetry` events in your Phoenix application.
+
+> `te·lem·e·try` - the process of recording and transmitting
+the readings of an instrument.
+
+As you follow along with this guide, you will be introduced
+to the building blocks of Telemetry (events and metrics),
+you will initialize a reporter that can show you events as
+they occur within your application, and we will guide you
+through the steps to properly instrument your own functions
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md]: Hi @samullen! I had not planned to get into writing a reporter in this guide, as this document started life as the LiveDashboard Telemetry guide :)

So this guide is really centered more around Metrics than it is Telemetry in general, although I wanted to give a clear picture of the ecosystem while still keeping this guide _reasonably_ high-level and _relatively_ easy to consume.

The overall idea being to 1) let users know there is Telemetry data readily available for consumption in their app, 2) there are purpose-built reporters that you can start using today, and 3) when you need custom instrumentation, here's a pattern you can follow.

File: guides/telemetry.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+to get a better idea of what makes up a telemetry event.
+Let's take a closer look.
+
+## Telemetry Events
+
+Many Elixir libraries (including Phoenix) are already using
+the `:telemetry` package (http://hexdocs.pm/telemetry) as a
+way to give users more insight into the behavior of their
+applications, by emitting events at key moments in the
+application lifecycle.
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md:22]: Thanks! ❤️

File: guides/telemetry.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md:23]: Done. I added some more basic information around `name`, `measurements`, and `metadata`.

File: guides/telemetry.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+This means that after each request, Phoenix, via telemetry,
+will emit a "stop" event, with a measurement of how long it
+took to get the response:
+
+```elixir
+:telemetry.execute([:phoenix, :endpoint, :stop], %{duration: duration}, %{conn: conn})
+```
+
+## Telemetry.Metrics
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md]: Done. Good note. 👍

File: guides/telemetry.md [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+
+## Reporters
+
+Reporters can subscribe to the Telemetry events you define,
+and use the common interface provided by `Telemetry.Metrics`,
+along with the measurements and metadata emitted in Telemetry
+events, to provide information meaningful to your
+application.
+
+### Phoenix.LiveDashboard
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md:136]: @samullen Yeah we got metrics! :D If you haven't had a chance to play with this yet, I'm very excited for you to check it out: https://hexdocs.pm/phoenix_live_dashboard/metrics.html

File: guides/telemetry.md [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+various measurements and metadata. With all of this data
+just waiting to be consumed, let's talk about reporters.
+
+## Reporters
+
+Reporters can subscribe to the Telemetry events you define,
+and use the common interface provided by `Telemetry.Metrics`,
+along with the measurements and metadata emitted in Telemetry
+events, to provide information meaningful to your
+application.
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md]: Done, thanks!

File: guides/telemetry.md [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+took to get the response:
+
+```elixir
+:telemetry.execute([:phoenix, :endpoint, :stop], %{duration: duration}, %{conn: conn})
+```
+
+## Telemetry.Metrics
+
+> Metrics are aggregations of Telemetry events with a specific name,
+providing a view of the system's behaviour over time. -- _`Telemetry.Metrics`_
```
[the remark this answers is not in the document]
Comment (mcrumm) [reply guides/telemetry.md]: Thank you @aaronrenner! :)

**Comment by mcrumm** [comment]
Thank you @samullen and @aaronrenner for all of your feedback ❤️ If I missed anything, or you see room for further improvement, please open a PR!

**Pull request merged**